Mathlib/Topology/ContinuousFunction/Bounded.lean

```diff
@@ -1512,6 +1514,122 @@ lemma norm_sub_nonneg (f : α →ᵇ ℝ) :
+
+@[inherit_doc]
+scoped[BoundedContinuousFunction] notation
+  "C_cb(" α ", " γ ")" => compactlySupported α γ
+
+lemma mem_compactlySupported {f : α →ᵇ γ} :
+    f ∈ C_cb(α, γ) ↔ HasCompactSupport f :=
+  TwoSidedIdeal.mem_mk' {z : α →ᵇ γ | HasCompactSupport z} .zero .add .neg' .mul_left .mul_right f
+
+lemma exist_norm_eq [c : Nonempty α] {f : α →ᵇ γ} (h : f ∈ C_cb(α, γ)) : ∃ (x : α),
+    ‖f x‖ = ‖f‖ := by
+  by_cases hs : (tsupport f).Nonempty
+  · obtain ⟨x, _, hmax⟩ := mem_compactlySupported.mp h |>.exists_isMaxOn hs <|
+      (map_continuous f).norm.continuousOn
+    refine ⟨x, le_antisymm (norm_coe_le_norm f x) (norm_le (norm_nonneg _) |>.mpr fun y ↦ ?_)⟩
+    by_cases hy : y ∈ tsupport f
+    · exact hmax hy
+    · simp [image_eq_zero_of_nmem_tsupport hy]
+  · suffices f = 0 by simp [this]
+    rwa [not_nonempty_iff_eq_empty, tsupport_eq_empty_iff, ← coe_zero, ← DFunLike.ext'_iff] at hs
+
+theorem norm_lt_iff_of_compactlySupported {f : α →ᵇ γ} (h : f ∈ C_cb(α, γ)) {M : ℝ}
+    (M0 : 0 < M) : ‖f‖ < M ↔ ∀ (x : α), ‖f x‖ < M := by
+  constructor
+  · intro hn x
+    exact lt_of_le_of_lt (norm_coe_le_norm f x) hn
+  · by_cases he : Nonempty α
+    · intro ha
+      obtain ⟨x, hx⟩ := exist_norm_eq h
+      rw [← hx]
+      exact ha x
+    · rw [not_nonempty_iff] at he
+      intro _
+      rw [norm_eq_zero_of_empty]
+      exact M0
+
+theorem norm_lt_iff_of_nonempty_compactlySupported [c : Nonempty α] {f : α →ᵇ γ}
+    (h : f ∈ C_cb(α, γ)) {M : ℝ} : ‖f‖ < M ↔ ∀ (x : α), ‖f x‖ < M := by
+  constructor
+  · intro hn x
+    exact lt_of_le_of_lt (norm_coe_le_norm f x) hn
+  · intro ha
+    obtain ⟨x, _⟩ := exists_true_iff_nonempty.mpr c
+    apply (norm_lt_iff_of_compactlySupported h (lt_of_le_of_lt (norm_nonneg (f x)) (ha x))).mpr ha
+
+theorem compactlySupported_eq_top_of_isCompact (h : IsCompact (Set.univ : Set α)) :
+    C_cb(α, γ) = ⊤ :=
+  eq_top_iff.mpr fun _ _ ↦ h.of_isClosed_subset (isClosed_tsupport _) (subset_univ _)
+
+theorem compactlySupported_eq_top [CompactSpace α] : C_cb(α, γ) = ⊤ :=
+  compactlySupported_eq_top_of_isCompact CompactSpace.isCompact_univ
+
+theorem compactlySupported_eq_top_iff [Nontrivial γ] :
+    C_cb(α, γ) = ⊤ ↔ IsCompact (Set.univ : Set α) := by
+  refine ⟨fun h ↦ ?_, compactlySupported_eq_top_of_isCompact⟩
+  obtain ⟨x, hx⟩ := exists_ne (0 : γ)
+  let f : α →ᵇ γ := const α x
+  have hf : f ∈ C_cb(α, γ) := by rw [h]; trivial -- missing `TwoSidedIdeal.mem_top` simp lemma
+  convert (mem_compactlySupported.mp hf).isCompact
+  rw [eq_comm, Set.eq_univ_iff_forall]
+  exact fun _ ↦ subset_tsupport _ <| by simpa [f]
+
+lemma hasCompactSupport_mul_of_continuous_compactlySupported (g : C(α, γ)) (f : C_cb(α, γ)) :
+    HasCompactSupport (g * f : C(α, γ)) := HasCompactSupport.mul_left
+  (mem_compactlySupported.mp f.2)
+
+lemma bounded_of_continuous_hasCompactSupport (g : C(α, γ)) (h : HasCompactSupport g) : ∃ (C : ℝ),
+    ∀ (x y : α), dist (g x) (g y) ≤ C := by
```

Comment:
I think (a), this should be more unbundled, and (b), it should just be a constructor for a bounded continuous function. Note: we leave `g` explicit so that this doesn't show up in the infoview as `ofCompactSupport _ _`.
```suggestion
lemma ofCompactSupport (g : α → γ) (hg₁ : Continuous g)
    (hg₂ : HasCompactSupport g) : α →ᵇ γ := by
```
Then it should be followed by a lemma like:
```lean
lemma ofCompactSupport_mem (g : α → γ) (hg₁ : Continuous g)
    (hg₂ : HasCompactSupport g) : ofCompactSupport hg₁ hg₂ ∈ C_cb(α, γ) := sorry
```